### DISCUSSION TITLE
Will built-in support for B2B Edition be considered?

### DISCUSSION THREAD

**Discussion by jp06** [bigcommerce/catalyst#544] [General]
Hi, just played with it a bit and this looks promising! The bootstrapped site also looks clean and performant. Great work on this!

I just wonder if built-in support for B2B Edition is planned or if it's out of scope. We've always gone the theme route when developing BC storefronts. Our projects tend to be B2B so we use B2B Edition. Still, we _might_ consider going headless and use this framework for B2C projects sometime in the future.

Thank you!

**Comment by becomevocal** [comment]
Great question @jp06! And kudos for being the first poster to our discussions here 👏👏👏 

Directionally, I'll note we want Catalyst to feel like it's a native extension of BC, just like the themes you are used to building. To hit that target, you can expect us to look at B2B Edition features.

For your next B2B build, would love feedback on what features would be most helpful to support.

**Reply by jp06** [reply]
@becomevocal - That's good to hear!

If the store has opt-in to BPE with theme storefront, you cannot customize the Buyer Portal. Even without Buyer Portal, you will still deal with scope issues sometimes when overriding JS stuff (but you can always host your own B2B script). So it is a big plus if it's easily customizable but this is probably a given with the nature of the project.

Regarding features, these are the ones usually used (first two I think are crucial):

1. Order History
2. Address Book
3. Buy Again
4. Shopping List
5. Quick Order Pad

However before that, B2B auth/roles stuff has to be implemented which might be harder (or not, not sure).

**Comment by bc-0dp** [comment]
This might be of interest to you @jp06 

https://support.bigcommerce.com/s/blog-article/aAn4v000000Tf53/buyer-portal-experience-for-existing-b2b-edition-users?language=en_US

https://github.com/B3BC/b2b-headless-example

**Reply by bc-john** [reply]
Hey @jp06 - 

This is all great feedback! We are well on our way to getting you full access to the Buyer Portal source code. I’ll make sure you’re on our list for early access. There is a fully integrated Catalyst variation in the works as well. In the meantime, as @bc-0dp  pointed out, we also have the new sample repo setup to help accelerate a headless deployment of the current Buyer Portal with Next.JS Commerce. Be sure to review the [setup guide](https://github.com/B3BC/b2b-headless-example/blob/main/docs/setup-guide.md) for specifics on enabling the headless channel in B2BE. You will need to do this on a B2BE Buyer Portal enabled sandbox. I'll touch base on slack as well.

**Reply by jpicalla-four13** [reply]
@bc-0dp  - Thank you for that example!
@bc-john - That's exciting, really appreciate that and thanks for the updates!

**Reply by marianela-gerard** [reply]
Hello! Thanks for these comments, we are also looking forward to the B2B integration in Catalyst. Is there an estimated date for it?
In the meantime we are reviewing the B2B-Headless example of next commerce, but the catalyst option is the one that really interests us.

**Comment by becomevocal** [comment]
💁 If you haven't seen the docs yet, we now have formal guidance including an integration branch for Catalyst + B2B Buyer Portal: https://www.catalyst.dev/docs/b2b
